api/src/main/resources/org/openmrs/liquibase/updates/liquibase-update-to-latest-2.1.x.xml

```diff
@@ -42,14 +42,14 @@
 	</changeSet>
 	<changeSet author="vshankar" id="201609171146-1">
 		<preConditions onFail="MARK_RAN">
-			<foreignKeyConstraintExists foreignKeyName="member_patient"/>
+			<foreignKeyConstraintExists foreignKeyTableName="cohort_member" foreignKeyName="member_patient"/>
```

Comment:
@k4pran Is this required or just a hedge against the future? If it's required, we just need to ensure that we make similar updates to the modules as well.

Reply:
> @k4pran Is this required or just a hedge against the future? If it's required, we just need to ensure that we make similar updates to the modules as well.

This would be required for integration tests to pass as some of them run through all the update xmls, liquibase now require the table to be referenced when setting up these constraints.

I didn’t realise other modules have update xmls but yes I think we should update them all. Do you know what modules will have them?

I guess another question, are there cases where clients using openmrs will have their own update xml changesets that we don’t have access to? I am just thinking if they have taken a snapshot that don’t have the foreignKeyTableName?

I also added a link to the breaking change in the PR description if you want to read a bit more about it

Reply:
> Do you know what modules will have them?

Well, at least the modules turned up [by this search](https://github.com/search?q=org%3Aopenmrs+foreignKeyConstraintExists&type=code)

> I guess another question, are there cases where clients using openmrs will have their own update xml changesets that we don’t have access to?

That's a real situation, but we can easily tell people who run into the issue how to fix it. Most modules (as opposed to core) rely on hand-maintained Liquibase scripts rather than snapshots.

> I also added a link to the breaking change in the PR description if you want to read a bit more about it

Ah, I was under the impression that that change was is Liquibase 4.11, not 4.7. I see.

Reply:
Yea liquibase 4.11 has a few additional breaking changes and also a few code changes we will need. I haven't looked deeply into the additional breaking changes. I will update the constraints for those other repos too. Should it be new tickets for each repo?

Reply:
@k4pran Yeah, most modules have their own separate Jira project, so we should stick to that.